app/screens/settings-screen/account/profile.tsx

```diff
@@ -2,36 +2,118 @@ import { ScrollView } from "react-native-gesture-handler"
 import { Screen } from "@app/components/screen"
 import { GaloyPrimaryButton } from "@app/components/atomic/galoy-primary-button"
 import { useI18nContext } from "@app/i18n/i18n-react"
-import { TouchableOpacity, View } from "react-native"
+import { ActivityIndicator, Button, TouchableOpacity, View } from "react-native"
 import { GaloyIcon } from "@app/components/atomic/galoy-icon"
-import { makeStyles, Text } from "@rneui/themed"
+import { makeStyles, Text, useTheme } from "@rneui/themed"
+import { usePersistentStateContext } from "@app/store/persistent-state"
+import { useNavigation } from "@react-navigation/native"
+import { StackNavigationProp } from "@react-navigation/stack"
+import { RootStackParamList } from "@app/navigation/stack-param-lists"
+import { useApolloClient, gql } from "@apollo/client"
+import { useUsernameLazyQuery } from "@app/graphql/generated"
+import { useEffect, useState } from "react"
+
+gql`
+  query username {
+    me {
+      username
+    }
+  }
+`
+
+type ProfileProps = {
+  username: string
+  token: string
+  selected?: boolean
+}
 
 export const ProfileScreen: React.FC = () => {
   const styles = useStyles()
+  const {
+    theme: { colors },
+  } = useTheme()
   const { LL } = useI18nContext()
+  const { persistentState } = usePersistentStateContext()
+  const navigation = useNavigation<StackNavigationProp<RootStackParamList>>()
 
-  const data = [
-    {
-      username: "User 1",
-      selected: true,
-    },
-    {
-      username: "User 2",
-      selected: false,
-    },
-    {
-      username: "User 3",
-      selected: false,
-    },
-  ]
+  const { galoyAuthToken: curToken, galoyAllAuthTokens: allTokens } = persistentState
+
+  const [profiles, setProfiles] = useState<ProfileProps[]>([])
+  const [fetchUsername, { error, refetch }] = useUsernameLazyQuery({
+    fetchPolicy: "no-cache",
+  })
+  const [loading, setLoading] = useState<boolean>(true)
+
+  useEffect(() => {
```

Comment:
I think we should cache the usernames - the loading of usernames take a long time. Can be done with more brainstorming in a separate PR